backend/apid/routers/checks.go

```diff
@@ -63,6 +68,95 @@ func (r *ChecksRouter) Mount(parent *mux.Router) {
 	parent.HandleFunc(path.Join(routes.PathPrefix, "{id}/execute"), r.adhocRequest).Methods(http.MethodPost)
 }
 
+// Patch the CheckConfig
+func (r *ChecksRouter) Patch(req *http.Request) (interface{}, error) {
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		return nil, actions.NewError(actions.InvalidArgument, err)
+	}
+	req.Body.Close()
+
+	var config corev2.CheckConfig
+	if err := json.Unmarshal(body, &config); err != nil {
+		return nil, actions.NewError(actions.InvalidArgument, err)
+	}
+
+	if len(config.RuntimeAssets) > 0 {
+		if missing, names := r.hasMissingAssets(req, config.RuntimeAssets); missing {
+			return nil, actions.NewError(actions.InvalidArgument, fmt.Errorf("runtime assets missing: %v", names))
+		}
+	}
+
+	req.Body = io.NopCloser(bytes.NewBuffer(body))
+	return r.handlers.PatchResource(req)
+}
+
+// Post the CheckConfig
+func (r *ChecksRouter) Post(req *http.Request) (interface{}, error) {
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		return nil, actions.NewError(actions.InvalidArgument, err)
+	}
+	req.Body.Close()
+
+	var config corev2.CheckConfig
+	if err := json.Unmarshal(body, &config); err != nil {
+		return nil, actions.NewError(actions.InvalidArgument, err)
+	}
+
+	if len(config.RuntimeAssets) > 0 {
+		if missing, names := r.hasMissingAssets(req, config.RuntimeAssets); missing {
+			return nil, actions.NewError(actions.InvalidArgument, fmt.Errorf("runtime assets missing: %v", names))
+		}
+	}
+
+	req.Body = io.NopCloser(bytes.NewBuffer(body))
+	return r.handlers.CreateResource(req)
+}
+
+// Put the CheckConfig
+func (r *ChecksRouter) Put(req *http.Request) (interface{}, error) {
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		return nil, actions.NewError(actions.InvalidArgument, err)
+	}
+	req.Body.Close()
+
+	var config corev2.CheckConfig
+	if err := json.Unmarshal(body, &config); err != nil {
+		return nil, actions.NewError(actions.InvalidArgument, err)
+	}
+
+	if len(config.RuntimeAssets) > 0 {
+		if missing, names := r.hasMissingAssets(req, config.RuntimeAssets); missing {
+			return nil, actions.NewError(actions.InvalidArgument, fmt.Errorf("runtime assets missing: %v", names))
+		}
+	}
+
+	req.Body = io.NopCloser(bytes.NewBuffer(body))
```

Comment:
This code block is repeated 3 times (in Patch, Post, Put). It might be worth isolating it in its own function.